mobile/library/cc/engine_builder.cc

```diff
@@ -914,7 +928,8 @@ std::unique_ptr<envoy::config::bootstrap::v3::Bootstrap> EngineBuilder::generate
     sink->mutable_typed_config()->PackFrom(metrics_config);
   }
 
-  if (!rtds_layer_name_.empty() && ads_api_type_.empty()) {
+  bootstrap->mutable_dynamic_resources();
```

Comment:
This adds an empty "dynamic resources" object to make it match the yaml, which has one.

Reply:
correct me if i'm wrong, but it seems to me like "dynamic_resources" only exists if CDS is enabled: https://github.com/envoyproxy/envoy/pull/25123/files#diff-46ea49cf2b9b992e3ec1e61b738a06149e6cd7bc00826bf7c034b3142b3db5efR438

Reply:
If you look at config.cc above, the tag "dynamic_resources:" is always in.  Just makes the design simpler since we now have two things which could get added to dynamic resources